fastapi/applications.py

```diff
@@ -93,6 +94,11 @@ def __init__(
             [] if middleware is None else list(middleware)
         )
         self.middleware_stack: ASGIApp = self.build_middleware_stack()
+        
+        def clear_lifetime_dependencies():
+            self.lifetime_dependencies = {}
```

Comment:
This stuff I'm a bit unclear about. I'm pretty sure the desired behavior is indeed for these dependencies to be re-initialized on every app startup, but I'm not sure if there's any precedent in the codebase for something like this. This is what I came up with. Open to ideas.